### PR TITLE
Fix incorrect sizing & pixelation of symbol previews on hidpi 

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes remaining issues with incorrect sizing and pixelation of preview icons for symbols, text formats, and other style objects

### CODE REVIEW

**Comment by nyalldawson** [comment]
Also note to self: there's an issue with the sizing of svg markers in one of these methods:

image

**Comment by nyalldawson** [comment]
No known outstanding issues now :+1:

File: src/core/symbology/qgssymbol.cpp [hunk 1]
```diff
@@ -1067,15 +1074,22 @@ QImage QgsSymbol::bigSymbolPreviewImage( QgsExpressionContext *expressionContext
...
+    p.drawLine( QLineF( 0, 50, 100, 50 ) );
+    p.drawLine( QLineF( 50, 0, 50, 100 ) );
   }
 
   QgsRenderContext context = QgsRenderContext::fromQPainter( &p );
   context.setFlag( Qgis::RenderContextFlag::RenderSymbolPreview );
   context.setFlag( Qgis::RenderContextFlag::Antialiasing );
   context.setFlag( Qgis::RenderContextFlag::HighQualityImageTransforms );
   context.setPainterFlagsUsingContext( &p );
+  context.setDevicePixelRatio( devicePixelRatio );
```
Comment (nirvn) [inline src/core/symbology/qgssymbol.cpp:1086]: Is that needed, you'll end up setting it a few lines down in updateRenderContextForScreen()

File: src/core/symbology/qgssymbol.h [hunk 2]
```diff
@@ -20,12 +20,15 @@
...
 #include "qgspropertycollection.h"
 #include "qgsrendercontext.h"
+#include "qgsscreenproperties.h"
 
 class QgsSymbolLayer;
 class QgsLegendPatchShape;
 class QgsSymbolRenderContext;
 class QgsLineSymbolLayer;
 
+class QScreen;
```
Comment (nirvn) [inline src/core/symbology/qgssymbol.h:30]: Do we need this here?

**Review by nirvn — COMMENTED** [review]
@nyalldawson , reviewing this now, the introduction of QgsScreenProperties was definitively the way to go, nice one.

File: src/core/symbology/qgssymbol.h (hunk 2 above)
Comment (nyalldawson) [reply src/core/symbology/qgssymbol.h:30]: nope, not anymore

**Review by nirvn — APPROVED** [review]

File: src/core/symbology/qgsstylemodel.cpp [hunk 3]
```diff
@@ -149,6 +138,14 @@ QVariant QgsStyleModel::data( const QModelIndex &index, int role ) const
             QString tooltip = QStringLiteral( "<h3>%1</h3><p><i>%2</i>" ).arg( name,
                               tags.count() > 0 ? tags.join( QLatin1String( ", " ) ) : tr( "Not tagged" ) );
 
+            // generate tooltips targetting the largest device pixel ratio for all attached screens
```
Comment (nyalldawson) [inline src/core/symbology/qgsstylemodel.cpp:141]:
```suggestion
            // generate tooltips for the largest device pixel ratio for all attached screens
```

**Pull request merged**